layers/+tools/tide/README.org

```diff
@@ -0,0 +1,92 @@
+#+TITLE: Tide Layer
+
+#+TAGS: layer|tool
+
+* Table of Contents                                       :TOC_4_gh:noexport:
+- [[#description][Description]]
+  - [[#features][Features:]]
+- [[#install][Install]]
+- [[#key-bindings][Key bindings]]
+  - [[#major-mode][Major Mode]]
+  - [[#reference-major-mode][Reference Major Mode]]
+
+* Description
+This layer installs [[https://github.com/ananthakumaran/tide][tide]] package which allows communication with [[https://github.com/Microsoft/TypeScript/wiki/Standalone-Server-%28tsserver%29][standalone
+typescript server]] =tsserver= for JavaScript/TypeScript development.
+
+** Features:
+  - First class support from =tsserver= just like =vscode=: speed and accuracy
+  - Linter 
+  - Refactor
+  - Go to definition
+  - Find references
+
+* Install
+See [[https://github.com/ananthakumaran/tide][tide]] for details. First you must have `tsserver` installed in your project
+or globally:
+
+#+begin_src sh
+  npm install -g tsserver
+#+end_src
+
+Next enable this layer in your =~/.spacemacs=. You will need to add =tide= to the
+existing =dotspacemacs-configuration-layers= list in this file. Then set the
+backend variables for JavaScript layer, React layer and TypeScript layer to =tide=
+in =dotspacemacs-configuration-layers=
+
+#+begin_src elsip
+  (javascript :variables
+               javascript-backend 'tide)
+
+  (typescript :variables
+              typescript-backend 'tide)
+#+end_src
+
+Both =javascript-backend= and =typescript-backend= can be set per project.
+
+When using this layer, make sure [[http://www.typescriptlang.org/docs/handbook/tsconfig-json.html][tsconfig.json]] or [[https://code.visualstudio.com/docs/languages/jsconfig][jsconfig.json]] is present in
+the root folder of the project.
+
+Example from emacs tide package: For JavaScript projects you may want to create
+=jsconfig.json= in the root folder of your project. =jsconfig.json= is =tsconfig.json=
+with allowjs attribute set to true.
+
+#+begin_src 
+  {
+    "compilerOptions": {
+      "target": "es2017",
+      "allowSyntheticDefaultImports": true,
+      "noEmit": true,
+      "checkJs": true,
+      "jsx": "react",
+      "lib": [ "dom", "es2017" ]
+    }
+  }
+#+end_src
+
+Without this file =tsserver= will pick up current folder as project root.
+* Key bindings
+** Major Mode
+| Key binding | Description                                     |
+|-------------+-------------------------------------------------|
+| ~SPC m = t~   | format buffer                                   |
```

Comment:
Same here with the binding, it should be `SPC m =` independent of the module which does the formating.